minecode_pipelines/miners/cargo.py

```diff
@@ -6,132 +6,66 @@
 # See https://github.com/aboutcode-org/purldb for support or download.
 # See https://aboutcode.org for more information about nexB OSS projects.
 #
-from datetime import datetime
 
-from minecode_pipelines.pipes import get_checkpoint_from_file
-from minecode_pipelines.pipes import get_commit_at_distance_ahead
-from minecode_pipelines.pipes import update_checkpoints_in_github
-from minecode_pipelines.pipes import get_changed_files
 from minecode_pipelines.pipes.cargo import store_cargo_packages
-from scanpipe.pipes.federatedcode import commit_changes
-from scanpipe.pipes.federatedcode import push_changes
-from minecode_pipelines import VERSION
-
+from scanpipe.pipes.federatedcode import commit_and_push_changes
 import json
 from pathlib import Path
 
-
-PACKAGE_BATCH_SIZE = 500
-COMMIT_BATCH_SIZE = 10
-
+BATCH_SIZE = 1000
 CARGO_CHECKPOINT_PATH = "cargo/checkpoints.json"
 
 
-def process_cargo_packages(cargo_index_repo, cloned_data_repo, config_repo, logger):
+def process_cargo_packages(cargo_index_repo, cloned_data_repo, logger):
     """
     Process Cargo index files commit by commit.
     Push changes to fed_repo after:
     - every `commit_batch` commits, OR when reaching HEAD.
     """
 
     base_path = Path(cargo_index_repo.working_tree_dir)
-
-    while True:
-        cargo_checkpoints = get_checkpoint_from_file(
-            cloned_repo=config_repo, path=CARGO_CHECKPOINT_PATH
-        )
-
-        checkpoints_last_commit = cargo_checkpoints.get("last_commit")
-
-        try:
-            next_commit = get_commit_at_distance_ahead(
-                cargo_index_repo,
-                checkpoints_last_commit,
-                num_commits_ahead=COMMIT_BATCH_SIZE,
-                branch_name="master",
+    file_counter = 0
+    purl_files = []
+    purls = []
+
+    for file_path in base_path.rglob("*"):
```

Comment:
Also we should use `LoopProgress` when mining purls from resource for proper progress indication.